frontend/cvat/src/app/dashboard/dashboard.component.ts

```diff
@@ -0,0 +1,140 @@
+import { Component, OnInit, Input, ComponentRef, ComponentFactoryResolver, OnDestroy,
+ViewContainerRef, ViewChild} from '@angular/core';
+import {MatDialog, MatDialogConfig } from '@angular/material/dialog';
+import { TaskConfigurationModalComponent } from '../task-configuration-modal/task-configuration-modal.component';
+import { DashboardService } from '../dashboard.service';
+import { Task } from '../models/task/task';
+import { DashboardItemComponent } from '../dashboard-item/dashboard-item.component';
+import { AnnotationFormat } from '../models/annotation-formats/annotation-format';
+import { forkJoin } from 'rxjs';
+
+@Component({
+  selector: 'app-dashboard',
+  templateUrl: './dashboard.component.html',
+  styleUrls: ['./dashboard.component.css']
+})
+export class DashboardComponent implements OnInit {
+
+  @ViewChild('taskView', {static: true, read: ViewContainerRef}) vc: ViewContainerRef;
+  title='CVAT Dashboard';
+  metaData: Object; //SHOULD THIS BE PRIVATE?
+  taskRef: ComponentRef<any>[]=[];
+
+  constructor(private matDialog: MatDialog, private dashboardService: DashboardService,
+  private CFR: ComponentFactoryResolver) { }
+
+  ngOnInit() {
+
+    forkJoin(
+      this.dashboardService.getAnnotationFormats(),
+      this.dashboardService.getTasks(),
+    ).subscribe( results=>{
+      for(let task of results[1]){
+        const componentFactory=this.CFR.resolveComponentFactory(DashboardItemComponent);
+        const componentRef=this.vc.createComponent(componentFactory);
+        componentRef.instance.task=task;
+        componentRef.instance.annotationFormats=results[0];
+        componentRef.instance.compInteraction=this;
+        this.taskRef.push(componentRef);
+      }
+    }
+  );
+}
+
+  dashboardCreateTaskButton() {
+  /*  const dialogConfig = new MatDialogConfig();
+    this.matDialog.open(TaskConfigurationModalComponent);
+    */
+    const dialogRef = this.matDialog.open(TaskConfigurationModalComponent, {
+     width: '500px',
```

Comment:
Avoid mixing CSS in your TypeScript files

Reply:
I can't remove this without distorting the modal that's being opened. I tried putting the CSS for width in the TaskConfigurationModalComponent's CSS, but it had no effect.